catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchSource.java

```diff
@@ -887,7 +944,8 @@ public String getUsername() {
 
   public void setUsername(String username) {
     this.username = username;
-    factory = null;
+    updateFactory();
+    updateScheduler();
```

Comment:
the downside of not using component-managed properties is that every setter is called when properties are updated through the administration ui. the result is that we do a lot of wasted work creating factories and schedulers that are never really needed (only the last setter that is called matters). however, I still prefer the straightforward style of container managed properties.